src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -489,8 +496,9 @@ def generate_config(
         if self.mgr.secure_monitoring_stack:
             r2: Dict[str, Any] = {'files': {}}
             for url, credentials in clusters_credentials.items():
-                r2['files'][f'prometheus_{url}_cert.crt'] = credentials['certificate']
-                credentials['cert_file_name'] = f'prometheus_{url}_cert.crt'
+                unique_id = str(uuid.uuid4())
+                r2['files'][f'prometheus_{unique_id}_cert.crt'] = credentials['certificate']
```

Comment:
if it's just that we need unique strings for each one and don't care what they actually are (which seems to be the case given we're generating it at random) why not just iterate over the integers `prometheus_1_cert`, `prometheus_2_cert` etc.?

Reply:
Makes sense, updated to use incremental integers now